neuro_san/client/direct_agent_session_factory.py

```diff
@@ -94,7 +94,8 @@ def create_session(self, agent_name: str, use_direct: bool = False,
 
         # DEF - We could do max_lifetime here, but waiting until that seems necessary.
         reservationist = DirectAgentReservationist(set([self.network_storage_dict.get("temp")]))
-        invocation_context = SessionInvocationContext(factory,
+        invocation_context = SessionInvocationContext(agent_name,
+                                                      factory,
```

Comment:
Add the agent name to the invocation context.  This makes for a better top-level trace name in something like LangSmith.